### DISCUSSION TITLE
Building on s390x using qemu

### DISCUSSION THREAD

**Discussion by thoni56** [cgreen-devs/cgreen#324] [Show and tell]
In the process of hunting down some bigendian issues on s390x @gavin09 mentioned having ran the build in qemu. I tried following some of the guides I found online, but did not manage to get it to boot properly.

It would be very helpful if we could have some help in setting that up here.

**Comment by gavin09** [comment]
Debian provides various tools for packaging. This link introduce some useful tools for cross compiling: https://wiki.debian.org/CrossCompiling.

The tool I use is pbuilder. However, I see some issues during this time. The test result is different compared to a real hardware. I got kernel core dumped fault when running test case 7 and test case 12 in s390x. I have not dig it out yet.

**Reply by gavin09** [reply]
From my point of view, I think Travis CI is good enough : )

**Reply by thoni56** [reply]
I was thinking about debugging. I needed three or four commits, which I since squashed, to understand the problem and fix it. If I had an environment in which to compile and debug, it would have been smoother. (And I like to try out stuff that I haven't used before ;-)

**Reply by gavin09** [reply]
I am happy to share anything if you encounter any issues : )

**Reply by thoni56** [reply]
I first tried in WSL, but that took ages, and finally looked like it hang, that's why I asked. But I just retried the live ISO in MSYS2 MingW64, which of course also took ages, but finally allowed me to install, and after waiting another eternity, it actually runs.

The guide I used was https://ubuntu.com/server/docs/virtualization-qemu but skipped all things with peripherals and I/0 and used `-nographic`.
